bip32/src/xkeys.rs

```diff
@@ -69,6 +69,12 @@ pub struct XPriv {
     pub(crate) xkey_info: XKeyInfo,
 }
 
+impl PartialEq for XPriv {
+    fn eq(&self, other: &XPriv) -> bool {
+        self.key.to_bytes() == other.key.to_bytes()
```

Comment:
better to implement using the key fingerprint and the xkeyinfo rather than the key itself :)